source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -48,7 +44,15 @@ bool ProcessorState::handleHeadersResponse(const HeadersResponse& response,
       return true;
     }
 
-    // If we got here, then the processor doesn't care about the body, so we can just continue.
+    if (send_trailers_ && filter_state_ >= FilterState::Trailers) {
```

Comment:
Seems fragile if we have future reordering/additions to do >=.

Reply:
I think you're referring to the >=, right? Let me review all the logic again and see if two or three bool flags might be easier to understand than one enum.

Reply:
I had to do all that ordering stuff to understand the state transitions and now that I figured that out we can get this done with a much simpler state mechanism, so PTAL the latest state of the code.